.circleci/config.yml

```diff
@@ -25,6 +25,17 @@ jobs:
       - checkout
       - run: make build
 
+  check-dependencies:
+    executor:
+      name: default
+    steps:
+      - checkout
+      - run:
+          name: Check unused Go dependencies
+          command: |
+            go mod tidy
+            if [[ -n $(git status --porcelain) ]]; then echo "There are unused dependencies that should be removed. Please execute `go mod tidy` to fix it."; exit 1; fi
```

Comment:
Nit pick: That check might cause false positives if any other file was modified and also doesn't show what the diff is. What do you think about using `git --no-pager diff --exit-code go.mod go.sum`?

Reply:
No problem including that to see the diff. The other problem is not going to happen because every step in circleci is running in an isolated environment without possible changes as you can see [here](https://app.circleci.com/pipelines/github/mattermost/mattermost-mattermod/681/workflows/10659967-533e-4822-b948-ae6961b1eef0/jobs/1684/steps)